tests/cstor/cspc/positive/provisioning/provisioning_test.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2019 The OpenEBS Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioning
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	cspc_v1alpha1 "github.com/openebs/maya/pkg/cstor/poolcluster/v1alpha1"
+	"github.com/openebs/maya/tests/artifacts"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("[CSPC] CSTOR STRIPE POOL PROVISIONING AND RECONCILIATION ", func() {
+	provisioningAndReconciliationTest(createCSPCObjectForStripe)
+})
+var _ = Describe("[CSPC] CSTOR MIRROR POOL PROVISIONING AND RECONCILIATION ", func() {
+	provisioningAndReconciliationTest(createCSPCObjectForMirror)
+})
+var _ = Describe("[CSPC] CSTOR RAIDZ POOL PROVISIONING AND RECONCILIATION ", func() {
+	provisioningAndReconciliationTest(createCSPCObjectForRaidz)
+})
+var _ = Describe("[CSPC] CSTOR RAIDZ2 POOL PROVISIONING AND RECONCILIATION ", func() {
+	provisioningAndReconciliationTest(createCSPCObjectForRaidz2)
+})
+
+func provisioningAndReconciliationTest(createCSPCObject func())  {
```

Comment:
Good one @sonasingh46 :+1: 
It would be good idea if we can create issues for BDD testcases which need to be covered.
Please add comments, so that new contributors would be able to add testcases